src/ui/render.ts

```diff
@@ -177,36 +179,27 @@ export default class UI {
 
     inputDom.onfocus = () => {
       this.removeGroupSelection();
-
+      this.removeCellRangeSelection();
       cellDom.classList.add("lightsheet_table_selected_cell");
-
-      let columnIndex: number | undefined;
-      let rowIndex: number | undefined;
-
-      const cellIdInfo = this.checkCellId(cellDom);
-      if (!cellIdInfo) return;
-      const { keyParts, isIndex } = cellIdInfo;
-
-      if (isIndex) {
-        columnIndex = Number(keyParts[0]);
-        rowIndex = Number(keyParts[1]);
-      } else {
-        const columnKey = cellDom.id;
-        const rowKey = cellDom.parentElement?.id;
-
-        columnIndex = this.lightSheet.sheet.getColumnIndex(
-          generateColumnKey(columnKey!),
-        );
-        rowIndex = this.lightSheet.sheet.getRowIndex(generateRowKey(rowKey!));
+      const { columnIndex, rowIndex } = this.getColumnAndRowIndex(cellDom);
+      if (columnIndex !== undefined && rowIndex !== undefined) {
+        this.selectedCell.push(columnIndex, rowIndex);
       }
-      this.selectedCell?.push(Number(columnIndex), Number(rowIndex));
     };
 
     inputDom.onblur = () => {
       this.selectedCell = [];
       cellDom.classList.remove("lightsheet_table_selected_cell");
     };
 
+    inputDom.onmousedown = (e: MouseEvent) => {
+      this.handleMouseDown(e, cellDom);
+    };
+
+    inputDom.onmouseup = (e: MouseEvent) => {
+      this.handleMouseUp(e, cellDom);
+    };
```

Comment:
Shouldn't we use `onmouseover` instead of `onmouseup` here? Other spreadsheet applications actively update the selection when you hold the mouse button down and hover other cells.